griptape/utils/decorators.py

```diff
@@ -24,8 +25,28 @@ def activity(config: dict) -> Any:
 
     def decorator(func: Callable) -> Any:
         @functools.wraps(func)
-        def wrapper(self: Any, *args, **kwargs) -> Any:
-            return func(self, *args, **kwargs)
+        def wrapper(self: Any, params: dict) -> Any:
+            func_params = inspect.signature(func).parameters.copy()
+            func_params.pop("self")
+
+            kwarg_var = None
+            for param in func_params.values():
+                if param.kind == inspect.Parameter.VAR_KEYWORD:
+                    kwarg_var = func_params.pop(param.name).name
+                    break
+
+            kwargs = {k: v for k, v in params.get("values", {}).items() if k in func_params or kwarg_var is not None}
+
+            if "params" in func_params or kwarg_var is not None:
+                kwargs["params"] = params
+            if "values" in func_params or kwarg_var is not None:
+                kwargs["values"] = params.get("values")
+
+            for param_name in func_params.keys():  # noqa: SIM118
+                if param_name not in kwargs:
+                    kwargs[param_name] = None
+
+            return func(self, **kwargs)
```

Comment:
Can you please move this out to a function and add a test?

Reply:
cov-bot says im good tho

Reply:
but really we dont have decorators.py tests. i added a few methods to `test_base_tool`

Reply:
> cov-bot says im good tho

That guy will approve anything